src/main/java/jenkins/branch/BranchStatusColumn.java

```diff
@@ -104,4 +106,19 @@ public boolean shownByDefault() {
             return false;
         }
     }
+
+    /**
+     * Hide this column from user views as it would only confuse them.
+     */
+    @Extension
+    public static class DescriptorVisibilityFilterImpl extends DescriptorVisibilityFilter {
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public boolean filter(Object context, Descriptor descriptor) {
+            return !(descriptor instanceof DescriptorImpl) || context instanceof MultiBranchProjectViewHolder.ViewImpl;
```

Comment:
Might be clearer if written as

```java
return !(descriptor instanceof DescriptorImpl &&
         !(context instanceof MultiBranchProjectViewHolder.ViewImpl));
```

or even

```java
if (descriptor instanceof DescriptorImpl &&
    !(context instanceof MultiBranchProjectViewHolder.ViewImpl)) {
    return false;
} else {
    return true;
}
```

Of course clearer and more efficient still would have been if `@Extension` had never been invented and SezPoz had been used as intended (`-source 8` for brevity):

```java
@DescriptorVisibilityFilterRegistration(type=DescriptorImpl.class) public static final DescriptorVisibilityFilter HIDE_ME_ELSEWHERE =
    (context, descriptor) -> context instanceof MultiBranchProjectViewHolder.ViewImpl;
```